### PR TITLE
[usrp] init

### PR DESCRIPTION
**Pull request by sjkelly**
This adds library support for the USRP series of SDRs by NI. 

https://github.com/EttusResearch/uhd

### CODE REVIEW

File: U/USRPHardwareDriver/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+# TODO: Windows has several issues with boost threads. There is a WIP branch:
+# https://github.com/JuliaTelecom/uhd/tree/juliatelecom/patch-v4.1.0.1
+platforms = filter!(p -> !Sys.iswindows(p) && !in(arch(p),("armv7l","armv6l")), supported_platforms(;experimental=true))
+platforms = expand_cxxstring_abis(platforms)
+# For some reasons, building for CXX03 string ABI doesn't actually work, skip it
```
Comment (staticfloat) [inline U/USRPHardwareDriver/build_tarballs.jl:43]: This is potentially problematic. What is the error?

Comment (sjkelly) [reply U/USRPHardwareDriver/build_tarballs.jl]: I added another commit removing it. I get the warning typically with Boost telling me to add it: https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=13355&view=results 

Though it looks like it expanded to `cxx11` automatically?

File: U/USRPHardwareDriver/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+name = "USRPHardwareDriver"
+version = v"4.1.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/EttusResearch/uhd.git", "f633b497ff931c3c180d00148fc66e4feb8b8b6a")
+]
+
+dependencies = [
+    Dependency("boost_jll"; compat="=1.76.0"), # max gcc7
```
Comment (staticfloat) [inline U/USRPHardwareDriver/build_tarballs.jl:14]: Out of curiosity, why did you specify such a strict `boost_jll` compat version?

Comment (sjkelly) [reply U/USRPHardwareDriver/build_tarballs.jl:14]: Mentioned here: https://github.com/JuliaPackaging/Yggdrasil/issues/3024

Full version in the SONAME on linux.

File: U/USRPHardwareDriver/build_tarballs.jl (hunk 1 above)
Comment (staticfloat) [reply U/USRPHardwareDriver/build_tarballs.jl]: I built locally, and I can expand to both `cxx03` and `cxx11` no problem, so I pushed a commit here to test in on Yggdrasil.

**Pull request merged**